src/pages/TripBookingPage/PassengerDetailsForm/PassengerDetailsForm.tsx

```diff
@@ -0,0 +1,165 @@
+import { Formik, Form, FieldArray, FastField, FormikProps } from 'formik';
+import { Box, Grid, MenuItem, Paper, Typography } from '@mui/material';
+import getValidationSchema from './validation';
+import { Select, TextField } from 'formik-mui';
+import { useTranslation } from 'react-i18next';
+import { FareDetails } from '../../../components/FareDetails/FareDetails';
+import { StyledButton } from '../../../components/Button/Button.styled';
+import { useState, useEffect, useRef } from 'react';
+import { useTheme } from '@emotion/react';
+
+interface IPassengerDetails {
+    passengers: {
+        seatNumber: number;
+        fullName: string;
+        age: string;
+        gender: string;
+    }[];
+}
+
+const PassengerDetailsForm = () => {
+    const { t } = useTranslation('passengerDetails');
+    const theme = useTheme();
+
+    const [languageChangeKey, setLanguageChangeKey] = useState(0);
+    const formikRef = useRef<FormikProps<IPassengerDetails>>(null);
+
+    useEffect(() => {
+        // Incrementing languageChangeKey to force re-render when language changes
+        setLanguageChangeKey((prevKey) => prevKey + 1);
+        void formikRef.current?.validateForm();
```

Comment:
is it required in useEffect/ ?

Reply:
I am using FastField component provided by Formik for better performance when number of form inputs increases.
When using FastField, form labels did not automatically change language when language state has changed.
Similarly for error message language to change, form validation has to run again.

So useEffect was used to trigger the re-render with new language.